includes/class-sensei-guest-user.php

```diff
@@ -12,6 +12,22 @@
 	exit; // Exit if accessed directly.
 }
 
+// Only works when added outside of class.
+// TODO Check the reason later.
+add_action( 'init', 'sensei_start_session' );
+
+/**
+ * Starts the session if already not started.
+ *
+ * @since $$next-version$$
+ * @access private
+ */
+function sensei_start_session() {
```

Comment:
I think we are running into the problem we originally wanted to avoid by creating real users: PHP sessions and cookies don't work well for some WordPress hosts and, at best, site owners need to configure an exception for our cookie. 

Could an approach work where we filter [`determine_current_user`](https://developer.wordpress.org/reference/hooks/determine_current_user/), and return false on non-course pages? We would keep the user session going, but during the request, the user would be considered not logged in.

Reply:
That's a very good idea @yscik ! I've [updated the code](https://github.com/Automattic/sensei/pull/6280/commits/83ef3990b69e0eb10059bcc3d3a35646a133cddb) to use `wp_set_current_user` to impersonate a logged-out user instead of actually logging out and using sessions. Thanks!